composer/trainer/trainer.py

```diff
@@ -1022,9 +1100,157 @@ def saved_checkpoints(self) -> List[Tuple[Timestamp, List[pathlib.Path]]]:
+
+        # Scale Schedule Ratio and Schedulers
+        if scale_schedule_ratio != 1.0:
+            self.state.max_duration = _scale_max_duration_by_ssr(scale_schedule_ratio, self.state.max_duration)
+        if schedulers is not None:
+            self.state.schedulers = _compile_schedulers(schedulers, self.state, scale_schedule_ratio)
+            self._step_schedulers_every_batch = _should_step_schedulers_every_batch(schedulers,
+                                                                                    step_schedulers_every_batch)
+        else:
+            if scale_schedule_ratio != 1.0:
+                warnings.warn("Specifying `scale_schedule_ratio` without `schedulers` has no effect.")
+
+            if step_schedulers_every_batch is not None:
+                warnings.warn("Specifying `step_schedulers_every_batch` without `schedulers` has no effect.")
+
+            if step_schedulers_every_batch is not None:
+                warnings.warn("Specifying `step_schedulers_every_batch` without `schedulers` has no effect.")
+
+        if len(ensure_tuple(schedulers)) == 0:
+            warnings.warn(f"NoSchedulerWarning: No schedulers were specified. The learning rate will be constant.")
+
+        # Evaluators
+        if eval_dataloader is not None:
+            self.evaluators = [
+                ensure_evaluator(evaluator, self._original_model.metrics(train=False))
+                for evaluator in ensure_tuple(eval_dataloader)
+            ]
+            _set_evaluator_interval_and_subset_num_batches(
+                evaluators=self.evaluators,
+                eval_interval=eval_interval,
+                subset_num_batches=eval_subset_num_batches,
+            )
+            if len(self.evaluators) == 0:
+                if eval_subset_num_batches != -1:
+                    warnings.warn("Specifying `eval_subset_num_batches` without an `eval_dataloader` has no effect.")
+                if eval_interval != 1:
+                    warnings.warn("Specifying `eval_interval` without an `eval_dataloader` has no effect.")
+
+        if len(self.evaluators) == 0:
+            warnings.warn(("No `eval_dataloader` was specified. Please specify `eval_dataloader` to periodically "
+                           "evaluate your model while training."))
+
+        # Grad Accum
+        if grad_accum is not None:
+            self.state.grad_accum, self.adaptive_gradient_accumulation = _get_grad_accum(grad_accum, self._device)
+
+        # Grad Clip Norm
+        if grad_clip_norm is not None:
+            if self.state.is_model_deepspeed:
+                raise ValueError(
+                    ("When using DeepSpeed, the `grad_clip_norm` must be set when constructing the . "
+                     f"{type(self).__name__} -- e.g. via {type(self).__name__}(grad_clip_norm=...). It cannot be "
+                     f"specified on {type(self).__name__}.{type(self).fit.__name__}()."))
```

Comment:
```suggestion
                raise ValueError("Changing the grad_clip_norm when using DeepSpeed is not supported.")
```

edited to be shorter and mirror the message below re: precision.

Reply:
Don't think we need the complexity of the handle the `type` and names etc?

Reply:
Fixed both!